tests/gold_tests/pluginTest/slice/slice_prefetch.test.py

```diff
@@ -0,0 +1,178 @@
+ps.ReturnCode = 0
+ps.Streams.stderr = "gold/slice_200.stderr.gold"
+ps.Streams.stdout.Content = Testers.ContainsExpression("200 OK", "expected 200 OK response")
+tr.StillRunningAfter = ts
+
+# 1 Test - Full object slice with nest 2 blocks prefetched in background, block bytes= 7
+tr = Test.AddTestRun("Test - Full object slice with nest 2 blocks prefetched in background, block bytes= 7")
+ps = tr.Processes.Default
+ps.Command = curl_and_args + ' http://sliceprefetch2bytes1/path'
+ps.ReturnCode = 0
+ps.Streams.stderr = "gold/slice_200.stderr.gold"
+ps.Streams.stdout.Content = Testers.ContainsExpression("200 OK", "expected 200 OK response")
+tr.StillRunningAfter = ts
+
+# 2 Test - Full object slice with only next block prefetched in background, block bytes= 5
+tr = Test.AddTestRun("Full object slice with only next block prefetched in background, block bytes= 5")
+ps = tr.Processes.Default
+ps.Command = curl_and_args + ' http://sliceprefetch1bytes2/path'
+ps.ReturnCode = 0
+ps.Streams.stderr = "gold/slice_200.stderr.gold"
+ps.Streams.stdout.Content = Testers.ContainsExpression("200 OK", "expected 200 OK response")
+tr.StillRunningAfter = ts
+
+# 3 Test - Full object slice with nest 2 blocks prefetched in background, block bytes= 5
+tr = Test.AddTestRun("Full object slice with nest 2 blocks prefetched in background, block bytes= 5")
+ps = tr.Processes.Default
+ps.Command = curl_and_args + ' http://sliceprefetch2bytes2/path'
+ps.ReturnCode = 0
+ps.Streams.stderr = "gold/slice_200.stderr.gold"
+ps.Streams.stdout.Content = Testers.ContainsExpression("200 OK", "expected 200 OK response")
+tr.StillRunningAfter = ts
+
+# 4 Test - Whole asset via range
+tr = Test.AddTestRun("Whole asset via range")
+ps = tr.Processes.Default
+ps.Command = curl_and_args + ' http://sliceprefetch1bytes1/path' + ' -r 0-'
+ps.ReturnCode = 0
+ps.Streams.stderr = "gold/slice_206.stderr.gold"
+ps.Streams.stdout.Content = Testers.ContainsExpression("206 Partial Content", "expected 206 response")
+ps.Streams.stdout.Content += Testers.ContainsExpression("Content-Range: bytes 0-17/18", "mismatch byte content response")
+tr.StillRunningAfter = ts
+
+# 5 Test - Non aligned slice request
+tr = Test.AddTestRun("Non aligned slice request")
+ps = tr.Processes.Default
+ps.Command = curl_and_args + ' http://sliceprefetch1bytes1/path' + ' -r 5-16'
+ps.ReturnCode = 0
+ps.Streams.stderr = "gold/slice_mid.stderr.gold"
+ps.Streams.stdout.Content = Testers.ContainsExpression("206 Partial Content", "expected 206 response")
+ps.Streams.stdout.Content += Testers.ContainsExpression("Content-Range: bytes 5-16/18", "mismatch byte content response")
+tr.StillRunningAfter = ts
+
+# 6 Test - special case, begin inside last slice block but outside asset len
+tr = Test.AddTestRun("Invalid end range request, 416")
+beg = len(body) + 1
+end = beg + block_bytes
+ps = tr.Processes.Default
+ps.Command = curl_and_args + ' http://sliceprefetch1bytes1/path' + ' -r {}-{}'.format(beg, end)
```

Comment:
we've been switching over to this format convention for autests:
f'http://sliceprefetch1bytes1/path -r {beg}-{end}'